src/main/groovy/edu/oregonstate/mist/cardsapi/resources/CardsResource.groovy

```diff
@@ -65,4 +83,165 @@ class CardsResource extends Resource {
+        }
+
+        if(!rarities) {
+            rarities = validRarities
+        } else {
+            List<String> invalidRarities = rarities - validRarities
+            if(invalidRarities) {
+                return badRequest("Invalid rarities: \'${invalidRarities.join("\', \'")}\'. " +
+                        "Valid rarities are: " +
+                        "basic, common, uncommon, rare").build()
+            }
+        }
+
+        if(!(name.or("")).matches(regEx)) {
+            return badRequest("Invalid name: \'" + name.get() +
+                    "\'. Name must match pattern: " +
+                    regEx).build()
+        }
+
+        for(int i = 0; i < keywords.size(); i++) {
+            if(!keywords[i].matches(regEx)) {
+                return badRequest("Invalid keyword: \'" + keywords[i] +
+                        "\'. All keywords " +
+                        "must match pattern: " + regEx).build()
+            }
+        }
+
+        List<Card> cards = cardFluent.getCards(types, name.orNull(), colors, rarities,
+                energyMin.or(0), energyMax.or(999),
+                keywords, number.or(10), isRandom.or(true))
+
+        ResultObject cardResult = cardsResult(cards)
+        ok(cardResult).build()
+    }
+
+    @POST
+    @Path('')
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    Response postCard (@Valid Card newCard) {
+
+        Response response = cardValidator(newCard)
+        if(response) {
+            return response
+        }
+
+        Integer id = cardDAO.getNextId()
+        cardDAO.postCard(
+                id,
+                newCard.type,
+                newCard.name,
+                newCard.color,
+                newCard.rarity,
+                newCard.energy,
+                newCard.description)
+
+        Card card = cardDAO.getCardById(id)
+        ResultObject cardResult = cardsResult(card)
+        created(cardResult).build()
+    }
+
+    @PUT
+    @Path('{id}')
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    Response putCard(@PathParam('id') IntParam id, @Valid Card updateCard) {
+
+        if(cardDAO.cardExists(id.get()) != 1) {
+            return notFound().build()
+        }
+        Response response = cardValidator(updateCard)
+        if(response) {
+            return response
+        }
+
+        cardDAO.putCard(id.get(), updateCard.type, updateCard.name,
+                updateCard.color, updateCard.rarity,
+                updateCard.energy, updateCard.description)
+        Card card  = cardDAO.getCardById(id.get())
+        ResultObject cardResult = cardsResult(card)
+        ok(cardResult).build()
+    }
+
+    @DELETE
+    @Path('{id}')
+    Response deleteCard(@PathParam('id') IntParam id) {
+
+        if(cardDAO.cardExists(id.get()) != 1) {
+            return notFound().build()
+        }
+
+        cardDAO.deleteCard(id.get())
+        Response.status(204).build()
```

Comment:
Let's use something similar to [here](https://github.com/osu-mist/slay-the-spire-cards-api/blob/cf7f066daf55e23c13df6b96e28c3a46b566128e/src/main/groovy/edu/oregonstate/mist/api/Resource.groovy#L64) instead of hardcoding `204`.